L2Tests/tests/SystemService_L2test.cpp

```diff
@@ -0,0 +1,454 @@
+         break;
+      }
+   }
+
+    signalled = m_event_signalled;
+    m_event_signalled = SYSTEMSERVICEL2TEST_STATE_TEST_EVENT_NONE;
+    
+    return signalled;
+}
+
+
+/**
+ * @brief Compare two request status objects
+ *
+ * @param[in] data Expected value
+ * @return true if the argument and data match, false otherwise
+ */
+MATCHER_P(MatchRequestStatus, data, "")
+{
+    bool match = true;
+    std::string expected;
+    std::string actual;
+
+    data.ToString(expected);
+    arg.ToString(actual);
+    TEST_LOG(" rec = %s, arg = %s",expected.c_str(),actual.c_str());
+    EXPECT_STREQ(expected.c_str(),actual.c_str());
+
+    return match;
+}
+
+/********************************************************
+************Test case Details **************************
+** 1. Get temperature from systemservice
+** 2. Set temperature threshold
+** 3. Temperature threshold change event triggered from IARM
+** 4. Verify that threshold change event is notified
+*******************************************************/
+
+TEST_F(Systemservice_L2test,SystemServiceGetSetTemperature)
+{
+    JSONRPC::LinkType<Core::JSON::IElement> jsonrpc(SYSTEM_CALLSIGN, L2TEST_CALLSIGN);
+    StrictMock<AsyncHandlerMock> async_handler;
+    uint32_t status = Core::ERROR_GENERAL;
+    JsonObject params,thresholds;
+    JsonObject result;
+    SystemServiceL2test_async_events_t signalled = SYSTEMSERVICEL2TEST_STATE_TEST_EVENT_NONE;
+    std::string message;
+   JsonObject expected_status;
+
+    /* Get temperature from systemservice*/
+    ON_CALL(iarmBusImplMock, IARM_Bus_Call)
+   .   WillByDefault(
+          [](const char* ownerName, const char* methodName, void* arg, size_t argLen) {
+              EXPECT_EQ(string(ownerName), string(_T(IARM_BUS_PWRMGR_NAME)));
+              EXPECT_EQ(string(methodName), string(_T(IARM_BUS_PWRMGR_API_GetThermalState)));
+               auto param = static_cast<IARM_Bus_PWRMgr_GetThermalState_Param_t*>(arg);
+               param->curTemperature = 100;
+               return IARM_RESULT_SUCCESS;
+     });
+
+    status = InvokeServiceMethod("org.rdk.System.1", "getCoreTemperature", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    EXPECT_TRUE(result["success"].Boolean());
+    EXPECT_STREQ("100.000000", result["temperature"].Value().c_str());
+
+    /* errorCode and errorDescription should not be set */
+    EXPECT_FALSE(result.HasLabel("errorCode"));
+    EXPECT_FALSE(result.HasLabel("errorDescription"));
+
+    /* Register for temperature threshold change event. */
+    status = jsonrpc.Subscribe<JsonObject>(JSON_TIMEOUT,
+                                           _T("onTemperatureThresholdChanged"),
+                                           &AsyncHandlerMock::onTemperatureThresholdChanged,
+                                           &async_handler);
+
+    EXPECT_EQ(Core::ERROR_NONE, status);
+
+    /*Set temperature threshold */
+
+    EXPECT_CALL(iarmBusImplMock, IARM_Bus_Call)
```

Comment:
Can we split the get and set into different tests.

Reply:
This was part of L2 sequence.. we can remove get API here as it is not making much difference

Reply:
ok sounds good.